modules/tensor_mechanics/src/materials/IsotropicPlasticityStressUpdate.C

```diff
@@ -175,25 +176,6 @@ IsotropicPlasticityStressUpdateTempl<false>::computeHardeningValue(const Real &
   return _hardening_variable_old[_qp] + _hardening_slope * scalar;
 }
 
-template <>
-ADReal
-IsotropicPlasticityStressUpdateTempl<true>::computeHardeningValue(const ADReal & scalar)
-{
-  if (_hardening_function)
-  {
-    const Real strain_old = this->_effective_inelastic_strain_old[_qp];
-    const Real t = strain_old + MetaPhysicL::raw_value(scalar);
-
-    DualReal hardening_function_value = _hardening_function->value(t);
-    hardening_function_value.derivatives() =
-        (strain_old + scalar).derivatives() * _hardening_function->timeDerivative(t);
-
```

Comment:
I can't see why we don't need to set `_hardening_function->value(strain_old + scalar).derivatives()` here.

Reply:
Because it's implemented in `Function.C`:
https://github.com/idaholab/moose/blob/next/framework/src/functions/Function.C#L50-L67

Reply:
👍